apps/backend/test/mocks/mocks.cash.spec.ts

```diff
@@ -31,4 +32,21 @@ describe('Tests the generated mock data', () => {
       expected.deposits.map((itm) => itm.amount)
     );
   });
+
+  it('should generate cash deposits with a date greater than the payment dates', () => {
+    const depositsDates = cashDepositsMock.map((itm) =>
+      parse(itm.deposit_date, 'yyyy-MM-dd', new Date())
+    );
+    const paymentsDates = aggregatedPayments(cashPaymentsMock).map((itm) =>
+      parse(itm.fiscal_close_date, 'yyyy-MM-dd', new Date())
+    );
+    // compareAsc returns -1 if the first date is before the second date, which causes this function to fail
+    expect(
+      depositsDates.every((depositDate) =>
+        paymentsDates.every(
```

Comment:
note: max number of deposits/payments is 30 and max number of deposits is 10 for the generated data